### PR TITLE
Google Floods Part 2: Adding tooltip graph; COUNTRY=cambodia

### PR DESCRIPTION
**Pull request by gislawill**
### Description

This adds graphing of flood levels by gauge for the 8 days before and predicted for two days after. This functionality was built based on the approach and UI originally used for EWS flood levels in Cambodia.

Ideas for design improvement:
- [ ] Show time better. Dates are shown well our data is much finer grained than this graph demonstrates
- [ ] Differentiate between past data and future predictions

## How to test the feature:
- [ ] Run in Mozambique
- [ ] Enable Google Flood layer and click a point

## Checklist - did you ...

Test your changes with
- [x] `REACT_APP_COUNTRY=rbd yarn start`
- [x] `REACT_APP_COUNTRY=cambodia yarn start`
- [x] `REACT_APP_COUNTRY=mozambique yarn start`
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

## Screenshot/video of feature:
Screenshot 2024-08-30 at 5 27 32 PM

### CODE REVIEW

File: api/app/googleflood.py [hunk 1]
```diff
@@ -15,6 +15,28 @@
...
+            logger.warning("Request failed at url %s: %s", url, e)
+            response_data = {}
+
+    if "error" in response_data:
+        logger.error("Error in response: %s", response_data["error"])
+        raise HTTPException(
+            status_code=500, detail="Error fetching data from Google API"
+        )
+
+    return response_data
```
Comment (gislawill) [inline api/app/googleflood.py:37]: A little abstraction

File: frontend/src/components/Common/Chart/index.tsx [hunk 2]
```diff
@@ -56,6 +56,7 @@ export type ChartProps = {
   showDownloadIcons?: boolean;
   iconStyles?: React.CSSProperties;
   downloadFilenamePrefix?: string[];
+  units?: string;
```
Comment (gislawill) [inline frontend/src/components/Common/Chart/index.tsx:59]: Adding optional units to the chart

File: frontend/src/components/Common/Chart/index.tsx [hunk 3]
```diff
@@ -292,6 +317,9 @@ const Chart = memo(
                   fontColor: '#CCC',
                   ...(config?.minValue && { suggestedMin: config?.minValue }),
                   ...(config?.maxValue && { suggestedMax: config?.maxValue }),
+                  maxTicksLimit: 8,
```
Comment (gislawill) [inline frontend/src/components/Common/Chart/index.tsx:320]: max of 8 ticks prevents the yaxis from getting too cluttered

File: frontend/src/components/Common/Chart/index.tsx (hunk 2 above)
Comment (gislawill) [reply frontend/src/components/Common/Chart/index.tsx:59]: We expect units to be translated before passing it into this component

File: frontend/src/components/MapView/MapTooltip/PointDataChart/PopupPointDataChart.tsx [hunk 4]
```diff
@@ -51,16 +53,17 @@ const PopupPointDataChart = memo(() => {
...
         <Chart
-          title={t(title)}
+          title={t(title, datasetParams)}
           config={config}
           data={dataset}
           xAxisLabel={
             isAdminBoundary(datasetParams)
               ? undefined
-              : t('Timestamps reflect local time in Cambodia')
+              : t('Timestamps reflect local time in region')
```
Comment (gislawill) [inline frontend/src/components/MapView/MapTooltip/PointDataChart/PopupPointDataChart.tsx:62]: Making this less custom. The translated values in Khmer still read Cambodia

**Comment by gislawill** [comment]
I'm seeing some unrelated errors in our suite:
500 response from https://api.earthobservation.vam.wfp.org/ows/?service=WCS&request=GetCoverage&version=2.0.0&coverageId=wp_pop_cicunadj&subset=Long(95.71,96.68)&subset=Lat(19.42,20.33)

I'll vcr record this test in the future once it's back up to prevent this flakiness

**Comment by gislawill** [comment]
Heads up @wadhwamatic, the graph design has been updated slightly to denote past vs future
Screenshot 2024-10-09 at 9 59 24 PM

File: frontend/src/components/MapView/MapTooltip/PointDataChart/PopupPointDataChart.tsx (hunk 4 above)
Comment (wadhwamatic) [reply frontend/src/components/MapView/MapTooltip/PointDataChart/PopupPointDataChart.tsx:62]: @gislawill - sorry, I didn't catch before that this change impacts the EWS-1294 flood layer as well. For that case, the previous text was more accurate. It illustrates to users that the timestamps are local to Cambodia. We're not adjusting based on users's location and it's not a generic region. This is how EWS-1294 reporting works

**Comment by wadhwamatic** [comment]
> Heads up @wadhwamatic, the graph design has been updated slightly to denote past vs future 

@gislawill - thanks! This looks good. A few requests to polish this a bit further:

1. Can we apply the label 'Now' or 'Today' to the vertical line? The same is shown in Flood Hub
2. Our legend shows a grey box for 'Unknown' measurements but in Flood Hub, the label is 'No Data'. Let's use the same terms if this is indeed describing the same measurement
3. Within the chart popup and in the legend for this layer, please add the text 'Unit of measure = discharge in m<sup>3/s</sup>'. Does the tooltip support markdown for the superscript text?

**Comment by gislawill** [comment]
> 1. Can we apply the label 'Now' or 'Today' to the vertical line? The same is shown in Flood Hub
> 2. Our legend shows a grey box for 'Unknown' measurements but in Flood Hub, the label is 'No Data'. Let's use the same terms if this is indeed describing the same measurement
> 3. Within the chart popup and in the legend for this layer, please add the text 'Unit of measure = discharge in m3/s'. Does the tooltip support markdown for the superscript text?

1. I originally had that but it was feeling way too crowded — the label would often cover the lines. I can give the graph a bit more space and try to fit that label in above the graph and below the title
2. Sure, we can update that copy. The description in the documentation makes it seem like "No data" isn't quite accurate but it's close enough and it is clearer
    > Screenshot 2024-10-10 at 9 42 37 PM
3. Sure, we can do that. It doesn't support markdown but we can use the unicode character ³ to add the superscript

**Comment by gislawill** [comment]
Remaining To Dos:

- [ ] Add label 'Today' to the vertical line in tooltop
- [ ] Update 'no data' to  'Unknown'
- [ ] Add the text 'Unit of measure = discharge in m3/s' to tooltip

**Comment by gislawill** [comment]
Heads up @wadhwamatic, I made the requested updates:
- [x] Add label 'Today' to the vertical line in tooltip
- [x] Update 'no data' to 'Unknown'
- [x] Add the text 'Unit of measure = discharge in m3/s' to tooltip


I did not add that 'Unit of measure = discharge in m3/s' to legend because there are no numeric values to see until the tooltip is opened. 
Legend:
Screenshot 2024-10-31 at 2 21 45 PM
New tooltip graph:
Screenshot 2024-10-31 at 2 19 11 PM

**Comment by gislawill** [comment]
I'm merging this into [feature/google-floods/1317](https://github.com/WFP-VAM/prism-app/tree/feature/google-floods/1317) (aka Part 1) to allow for one review

**Pull request merged**